analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -69,9 +66,15 @@ public sealed partial class MemberInitializerRedundant : SonarDiagnosticAnalyzer
 
         private void CheckInstanceMembers(SonarSyntaxNodeReportingContext c, TypeDeclarationSyntax declaration, IEnumerable<ISymbol> typeMembers)
         {
-            var typeInitializers = typeMembers.OfType<IMethodSymbol>().Where(x => x is { MethodKind: MethodKind.Constructor, IsStatic: false, IsImplicitlyDeclared: false }).ToList();
-            if (typeInitializers.Count == 0)
+            var constructors = typeMembers.OfType<IMethodSymbol>().Where(x => x is { MethodKind: MethodKind.Constructor, IsStatic: false }).ToList();
+            if (constructors.Exists(x =>
+                // Implicit parameterless constructor
+                x.IsImplicitlyDeclared
```

Comment:
It might be worth to have an extension method `x.IsPrimaryConstructor()`, or at least a method with a descriptive name (that would make your commentary obsolete).